src/site/content/en/blog/granular-chunking-nextjs/index.md

```diff
@@ -0,0 +1,273 @@
+
+## Introduction
+
+Like many web frameworks, Next.js and Gatsby use [webpack](https://webpack.js.org/) as their core
+bundler. webpack v3 introduced
+[`CommonsChunkPlugin`](https://webpack.js.org/plugins/commons-chunk-plugin/) to make it possible to
+output modules shared between different entry points in a single (or few) "commons" chunk. Shared
+code can be downloaded separately and stored in the browser cache early on which can result in a
+better loading performance.
+
+This pattern became popular with many single-page application frameworks adopting an entrypoint and
+bundle configuration that looked like this:
+
+<img class="w-screenshot w-screenshot--filled" src="./commons-pattern-diagram.png" alt="Common entrypoint and bundle configuration">
+
+Although practical, the concept of bundling all shared module code into a single chunk has its
+limitations. Modules not shared in every entry point can be downloaded for routes that do not use it
+resulting in more code being downloaded than necessary (`moduleC` and `page1` in the diagram). For
+this reason, along with a few others, webpack v4 removed the plugin in favor of a new
+one: [`SplitChunksPlugin`](https://webpack.js.org/plugins/split-chunks-plugin/).
+
+## Improved Chunking
+
+The default settings for `SplitChunksPlugin`  works well for most users. Multiple split chunks are
+created depending on a number of [conditions](https://webpack.js.org/plugins/split-chunks-plugin/#defaults)
+to prevent fetching duplicated code across multiple routes. 
+
+However, many web frameworks that use this plugin still follow a "single-commons" approach to chunk
+splitting. Next.js, for example, would generate a `commons` bundle that contained any module that is
+used in more than 50% of pages and all framework dependencies (`react`, `react-dom`, etc…).
+
+```javascript/11,16
+const splitChunksConfigs = {
+  //...
+  prod: {
+    chunks: 'all',
+    cacheGroups: {
+      default: false,
+      vendors: false,
+      commons: {
+        name: 'commons',
+        chunks: 'all',
+        minChunks: totalPages > 2 ? totalPages * 0.5 : 2,
+      },
+      react: {
+        name: 'commons',
+        chunks: 'all',
+        test: /[\\/]node_modules[\\/](react|react-dom|scheduler|use-subscription)[\\/]/,
+      },
+    },
+  },
+```
+Although including framework-dependent code into a shared chunk means that it can be downloaded and
+cached for any entrypoint, the usage-based heuristic of including common modules used in more than
+_half of pages_ isn't very effective. Modifying this ratio would only result in one of two outcomes:
+more unnecessary code gets downloaded or more code gets duplicated across multiple routes.
+
+To solve this problem, we adopted a [different
+configuration](https://github.com/zeit/next.js/pull/7696) for`SplitChunksPlugin` that reduces
+unecessary code for any route.
+
++   Any sufficiently large third-party module (> 160 KB) is split into its own individual chunk
```

Comment:
```suggestion
+   Any sufficiently large third-party module (greater than 160 KB) is split into its own individual chunk
```
The `>` always confuses me so I think readers will have to think less if we just use words instead